src/caffe/layers/data_layer.cpp

```diff
@@ -112,11 +118,12 @@ void DataLayer<Dtype>::load_batch(Batch<Dtype>* batch) {
   for (int item_id = 0; item_id < batch_size; ++item_id) {
       timer.Start();
       // Get a datum
-      Datum* datum = (reader_.full().pop("Waiting for data"));
+      string* str = (reader_.full().pop("Waiting for data"));
```

Comment:
Isn't it a different thing? I'm asking how you handle the case where e.g. your batch size is 32 and you have 1000 entries in your dataset. I'm actually trying to understand this in the context of a tiny error I noticed in the calculation of accuracy.